### PR TITLE
fix: change to draft only on changes requested

### LINKED ISSUE #64: Only draft the pull-request if a review was requested
All right then I will proceed with the following changes on the pull-request:
- the PR will be converted to draft only if there are reviews in "changes requested" state
- otherwise, only the reminder will be posted and no state change will occur on the pull-request

Sounds good?

_Originally posted by @gentlementlegen in https://github.com/ubiquity-os-marketplace/daemon-disqualifier/issues/59#issuecomment-2535819991_
            

Instead of always converting the PR to a draft, it should only happen if changes were requested by reviewers.[^01^]

[^01^]: ⚠ 51% possible duplicate - [Pull Request State Adjustments](https://www.github.com/ubiquity-os-marketplace/daemon-disqualifier/issues/59#59)

### PR DESCRIPTION
**Pull request by gentlementlegen**
Relates to https://github.com/ubiquity-os-marketplace/daemon-disqualifier/issues/59#issuecomment-2535819991
Resolves #64 

We turn the PR to draft only if changes were requested.

### CODE REVIEW

File: src/handlers/watch-user-activity.ts [hunk 1]
```diff
@@ -29,9 +29,9 @@ export async function watchUserActivity(context: ContextPlugin) {
     if (context.config.pullRequestRequired) {
       message.push(`- Be sure to link a pull-request before the first reminder to avoid disqualification.`);
     }
-    message.push(`- Reminders will be sent every \`${formatMillisecondsToDaysAndHours(context.config.warning / priorityValue)}\` if there is no activity.`);
+    message.push(`- Reminders will be sent every \`${formatMillisecondsToHumanReadable(context.config.warning / priorityValue)}\` if there is no activity.`);
     message.push(
-      `- Assignees will be disqualified after \`${formatMillisecondsToDaysAndHours(context.config.disqualification / priorityValue)}\` of inactivity.`
+      `- Assignees will be disqualified after \`${formatMillisecondsToHumanReadable(context.config.disqualification / priorityValue)}\` of inactivity.`
```
Comment (0x4007) [inline src/handlers/watch-user-activity.ts:34]: What happens on priority 0?

Comment (gentlementlegen) [reply src/handlers/watch-user-activity.ts:34]: A disaster most likely, what should we do on priority 0 for the deadline?

File: src/helpers/collect-linked-pulls.ts [hunk 2]
```diff
@@ -34,6 +34,7 @@ const query = /* GraphQL */ `
                   id: databaseId
                 }
               }
+              reviewDecision
```
Comment (whilefoo) [inline src/helpers/collect-linked-pulls.ts:37]: I guess this is based on the last review?

**Review by whilefoo — APPROVED** [review]

File: src/helpers/collect-linked-pulls.ts (hunk 2 above)
Comment (gentlementlegen) [reply src/helpers/collect-linked-pulls.ts:37]: According to the docs it contains the last global state of the review: https://docs.github.com/en/graphql/reference/enums#pullrequestreviewdecision

File: src/handlers/watch-user-activity.ts (hunk 1 above)
Comment (0x4007) [reply src/handlers/watch-user-activity.ts:34]: That or infinity. Let's try 1 for now then

Comment (gentlementlegen) [reply src/handlers/watch-user-activity.ts:34]: Even when the priority is `0` we still want the disqualifier to run right? If so I don't think infinity makes sense, it should use the same timers as a priority `1`.

**Pull request merged**